### PR TITLE
replace file constructors with context managers for python 3 compatibility

### LINKED ISSUE #66: ics_diff doesn't work under py3
I get this stack trace:
```
Traceback (most recent call last):
  File "/home/luc/.local/bin/ics_diff", line 11, in <module>
    load_entry_point('vobject==0.9.4.1', 'console_scripts', 'ics_diff')()
  File "/home/luc/.local/lib/python3.6/site-packages/vobject/ics_diff.py", line 194, in main
    cal1 = readOne(file(ics_file1))
NameError: name 'file' is not defined
```

As you can see I am useing python3.6 and vobject 0.9.4.1 (installed with pip).

### PR DESCRIPTION
**Pull request by wpercy**
`file()` was removed in Python 3 - replace all calls with `open()` and use with statements for closing. Resolves #66

### CODE REVIEW

File: vobject/change_tz.py [hunk 1]
```diff
@@ -64,9 +64,9 @@ def main():
         out_name = ics_file + '.converted'
         print("... Writing {0!s}".format(out_name))
 
-        out = file(out_name, 'wb')
-        cal.serialize(out)
-        print("Done")
+        with open(out_name, 'wb') as out:
+            cal.serialize(out)
+            print("Done")
```
Comment (htgoebel) [inline vobject/change_tz.py]: Arguable this line should not be part of the `with` block.

File: vobject/ics_diff.py [hunk 2]
```diff
@@ -191,11 +191,12 @@ def main():
...
         ignore_dtstamp = options.ignore
         ics_file1, ics_file2 = args
-        cal1 = readOne(file(ics_file1))
-        cal2 = readOne(file(ics_file2))
-        deleteExtraneous(cal1, ignore_dtstamp=ignore_dtstamp)
-        deleteExtraneous(cal2, ignore_dtstamp=ignore_dtstamp)
-        prettyDiff(cal1, cal2)
+        with open(ics_file1) as f, open(ics_file2) as g:
+            cal1 = readOne(f)
+            cal2 = readOne(g)
```
Comment (htgoebel) [inline vobject/ics_diff.py]: `f` and `g` are not used below, so the `with`-block should end here.

**Comment by lucc** [comment]
Any chance this will be released as 0.9.4.2 or 0.9.5? I saw that there are some milestones for the next releases but they are all done already.

**Comment by wpercy** [comment]
This will be released as part of 0.9.5

**Pull request merged**